usr/src/uts/common/fs/zfs/spa.c

```diff
@@ -6985,6 +6987,14 @@ spa_event_create(spa_t *spa, vdev_t *vd, const char *name)
 		}
 	}
 
+	if (hist_nvl) {
```

Comment:
I would actually prefer that as well, and only did it like this to match the style already in place.  If you expand the lines above you can see `if (vd) { ... }` already in place with almost the exact same logic as this block.